sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -199,27 +204,34 @@ def cache_manager(self):
     """Gets the cache manager held by current Interactive Environment."""
     return self._cache_manager
 
-  def set_pipeline_result(self, pipeline, result):
+  def set_pipeline_result(self, pipeline, result, is_main_job=True):
```

Comment:
Related to the default value of is_main_job, should that be a required parameter instead of having a default?

If these APIs are intended for internal use, removing default could make it more explicit. If it is for user consumption and users will be primarily concerned with the main jobs keeping the default makes sense.

Reply:
Thanks! They are for internal use, I'm removing the default values to make it explicit.